notebooks/algorithm-design/concepts.ipynb

```diff
@@ -0,0 +1,544 @@
+    "    aux_circ.measure_all()\n",
+    "    aux_circuits.append(aux_circ)\n",
+    "\n",
+    "\n",
+    "print(\"Original circuit:\")\n",
+    "display(original_circuit.draw(\"mpl\"))\n",
+    "for (circuit, pauli) in zip(aux_circuits, H.paulis):\n",
+    "    print(f\"Auxiliary circuit for {str(pauli)}\")\n",
+    "    display(circuit.draw(\"mpl\"))"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can now carry out the computation manually using `Sampler` and check the results on `Estimator`:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Sampler results:\n",
+      "  >> Expected value of X: 1.00000\n",
+      "  >> Expected value of Z: 0.00000\n",
+      "  >> Total expected value: 2.00000\n",
+      "Estimator results:\n",
+      "  >> Expected value of X: 1.00000\n",
+      "  >> Expected value of Z: 0.00000\n",
+      "  >> Total expected value: 2.00000\n"
+     ]
+    }
+   ],
+   "source": [
+    "from qiskit.primitives import Sampler, Estimator\n",
+    "from qiskit.circuit.library import IGate, ZGate\n",
+    "import numpy as np\n",
+    "\n",
+    "\n",
+    "## SAMPLER\n",
+    "sampler = Sampler()\n",
+    "job = sampler.run(aux_circuits)\n",
+    "probability_dists = job.result().quasi_dists\n",
+    "\n",
+    "expvals = []\n",
+    "for dist, pauli in zip(probability_dists, H.paulis):\n",
+    "    val = 0\n",
+    "    if str(pauli) == \"I\":\n",
+    "        Lambda = IGate().to_matrix().real\n",
+    "    else:\n",
+    "        Lambda = ZGate().to_matrix().real\n",
+    "    val += Lambda[0][0]*dist.get(0, 0)\n",
+    "    val += Lambda[1][1]*dist.get(1, 0)\n",
+    "    expvals.append(val)\n",
+    "\n",
+    "\n",
+    "print(\"Sampler results:\")\n",
+    "for (pauli, expval) in zip(H.paulis, expvals):\n",
+    "    print(f\"  >> Expected value of {str(pauli)}: {expval:.5f}\")\n",
+    "\n",
+    "total_expval = np.sum(H.coeffs*expvals).real\n",
+    "print(f\"  >> Total expected value: {total_expval:.5f}\")\n",
+    "\n",
+    "\n",
+    "## ESTIMATOR\n",
+    "observables = [*H.paulis, H]  # Note: run for individual Paulis as well as full observable H\n",
+    "\n",
+    "estimator = Estimator()\n",
+    "job = estimator.run([original_circuit]*len(observables), observables)\n",
+    "estimator_expvals = job.result().values\n",
+    "\n",
+    "print(\"Estimator results:\")\n",
+    "for (obs, expval) in zip(observables, estimator_expvals):\n",
+    "    if obs is not H:\n",
+    "        print(f\"  >> Expected value of {str(obs)}: {expval:.5f}\")\n",
+    "    else:\n",
+    "        print(f\"  >> Total expected value: {expval:.5f}\")\n"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Mathematical rigor (optional)\n",
```

Comment:
This entire section is probably unnecessary. Except maybe this: the expectation value is simply the weighted average of the values of the observable with respect to the probability of measuring the value. This is an incredibly intuitive result and maybe deserves a proof. 

The second fact that might be worth "proving" is that expectation values are linear. But this follows directly from properties of matrix multiplication (or the inner product more abstractly). 

I would recommend something like in the attached photo:
<img width="871" alt="image" src="https://user-images.githubusercontent.com/29473076/222588680-74cfde19-11b3-45e9-826a-9065dd9ffbcf.png">